code/game/machinery/canister_frame.dm

```diff
@@ -0,0 +1,92 @@
+#define FRAME_TIER_0	"0"
+#define FRAME_TIER_1	"1"
+#define FRAME_TIER_2	"2"
+//Canister Frames
+/obj/structure/canister_frame
+	name = "canister frame"
+	icon = 'icons/obj/atmos.dmi'
+	icon_state = "frame_0"
+	density = TRUE
+	var/mode_frame = FRAME_TIER_0
+
+/obj/structure/canister_frame/examine(user)
+	. = ..()
+
+/obj/structure/canister_frame/deconstruct(disassembled = TRUE)
+	if(!(flags_1 & NODECONSTRUCT_1))
+		new /obj/item/stack/sheet/metal(loc, 5)
+	qdel(src)
+
+/obj/structure/canister_frame/machine
+	name = "canister frame"
+	desc = "A frame used to build different kinds of canisters."
+
+/obj/structure/canister_frame/machine/examine(user)
+	. = ..()
+	if(mode_frame)
+		. += "<span class='notice'>This canister frame is tier [mode_frame].</span>"
+
+/obj/structure/canister_frame/machine/update_icon_state()
+	if (mode_frame == FRAME_TIER_0)
+		icon_state = "frame_0"
+	else if (mode_frame == FRAME_TIER_1)
+		icon_state = "frame_1"
+	else if (mode_frame == FRAME_TIER_2)
+		icon_state = "frame_2"
+
+/obj/structure/canister_frame/machine/attackby(obj/item/S, mob/user, params)
+	var/is_metal_sheet = istype(S, /obj/item/stack/sheet/metal)
+	var/is_plasteel_sheet = istype(S, /obj/item/stack/sheet/plasteel)
+	var/is_titanium_sheet = istype(S, /obj/item/stack/sheet/mineral/titanium)
+	var/is_bscrystal_sheet = istype(S, /obj/item/stack/sheet/bluespace_crystal)
+	var/is_plastitanium_sheet = istype(S, /obj/item/stack/sheet/mineral/plastitanium)
+	var/obj/item/stack/ST = S
```

Comment:
add an if check here, otherwise it will error if S is not a stack.

Reply:
`if(istype(S, var/obj/item/stack))
     return ..()`
but with a newline after the if